### ISSUE TITLE
Rector idea - Doctrine repositories

### ISSUE THREAD

**Issue by mssimi** [rectorphp/rector#365]
Hey what about rector that refactors usage of service locator to DI for doctrine repos.

Before:
```php
final class ProductController extends Controller{

  public function something(..): Response
  {
    $em = $this->getDoctrine()->getManager();
    $em->getRepository('SomethingBundle:Product')->findSomething(...);
    //...
  }

}
```

After:
```php
final class ProductController extends Controller{

  public function __construct(ProductRepository $productRepository)
  {
     $this->productRepository = $productRepository;
  }

  public function something(...): Response
  {
    $this->productRepository->findSomething(...);
    //...
  }

}
```

**Comment by TomasVotruba** [comment]
What about?
https://github.com/rectorphp/rector/pull/347

**Comment by mssimi** [comment]
Wanted to test it, but it does not work for me.

```
 [ERROR] Cannot autowire service                                                                                        
         "Rector\Rector\Architecture\RepositoryAsService\MoveRepositoryFromParentToConstructorRector": argument         
         "$entityForDoctrineRepositoryProvider" of method "__construct()" references interface                          
         "Rector\Contract\Bridge\EntityForDoctrineRepositoryProviderInterface" but no such service exists. Did you      
         create a class that implements this interface? in                                                              
         /home/mssimi/www/rector-fork/vendor/symfony/dependency-injection/Compiler/DefinitionErrorExceptionPass.php on  
         line 37  
```

**Comment by TomasVotruba** [comment]
Error is pretty clear on what you should do.

I see the case is a bit different. The name of issue confused me.

Could you prepare empty Rector with failing tests?

**Comment by mssimi** [comment]
> Error is pretty clear on what you should do.

Yep sorry you right, I was used to not provide any extra config to rectors, what about reading it's entity from entity annotation?

> Could you prepare empty Rector with failing tests?
Ok:)

**Comment by TomasVotruba** [comment]
Merged in #367 locally
